tests/test_butler.py

```diff
@@ -931,6 +931,9 @@ def runImportExportTest(self, storageClass):
         # Test that the repo actually has at least one dataset.
         datasets = list(exportButler.registry.queryDatasets(..., collections=...))
         self.assertGreater(len(datasets), 0)
+        # Add a DimensionRecord that's unused by those datasets.
+        skymapRecord = {"name": "example_skymap", "hash": (50).to_bytes(8, byteorder="little")}
+        exportButler.registry.insertDimensionData("skymap", skymapRecord)
         # Export those datasets.  We used TemporaryDirectory because there
         # doesn't seem to be a way to get the filename (as opposed to the file
         # object) from any of tempfile's temporary-file context managers.
```

Comment:
`tempfile.NamedTemporaryFile` ? Also, what does transfer-on-exist mean below?

Reply:
I was vaguely hoping you knew what either of these meant when I saw them.  `git blame` has proved my guilt on both.  I'm just going to remove them, because I don't think they're helpful - nobody knows what `transfer-on-exist` means anymore, and I don't see an obvious way to replace temporary directory usage with temporary file usage here (or why that would be better).